docs/content/docs/dev/datastream/fault-tolerance/checkpointing.md

```diff
@@ -292,4 +292,25 @@ The final checkpoint would be triggered immediately after all operators have rea
 without waiting for periodic triggering, but the job will need to wait for this final checkpoint 
 to be completed.
 
+## Unify file merging mechanism for checkpoints
+
+The unified file merging mechanism for checkpointing is introduced to Flink 1.20 as an MVP ("minimum viable product") feature, 
+which allows scattered small checkpoint files to be written into a single file, reducing the number of file creations 
+and file deletions, helping to alleviate the pressure of file system metadata management and file flooding problem. 
+The unified fie merging mechanism can be enabled by setting the property `state.checkpoints.file-merging.enabled` to `true`.
+**Note** that enabling this mechanism may lead to space amplification, that is, the actual occupation on the file system
+will be larger than actual state size. `state.checkpoints.file-merging.max-space-amplification` 
+can be used to limit the upper bound of space amplification.
+
+This mechanism is applicable to keyed state, operator state and channel state in Flink. Subtask level granular merging is 
+provided for shared scope state; TaskManager-level granular merging is provided for private scope state. The maximum number of subtasks
+allowed to be written to a single file can be configured through the `state.checkpoints.file-merging.max-subtasks-per-file` option.
+
+The unified fie merging mechanism also supports file merging across checkpoints, which can be enabled by setting
+`state.checkpoints.file-merging.across-checkpoint-boundary` to `true`.
+
+This mechanism introduces a file pool to handle concurrent writing scenarios. The blocking mode can be
```

Comment:
```suggestion
This mechanism introduces a file pool to handle concurrent writing scenarios. There are two modes....... The blocking mode...... while the non-blocking modes...... . This can be configured via ``.
```
Add some description to mode? instead of talking about enabling the option.